test/Microsoft.AspNetCore.Sockets.Common.Tests/Internal/Formatters/TextMessageFormatterTests.cs

```diff
@@ -0,0 +1,79 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using Microsoft.AspNetCore.Sockets.Internal.Formatters;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Sockets.Tests.Internal.Formatters
+{
+    public class TextMessageFormatterTests
+    {
+        [Fact]
+        public void WriteMultipleMessages()
+        {
+            const string expectedEncoding = "0:B:;14:T:Hello,\r\nWorld!;1:C:A;12:E:Server Error;";
+            var messages = new[]
+            {
+                MessageTestUtils.CreateMessage(new byte[0]),
+                MessageTestUtils.CreateMessage("Hello,\r\nWorld!",MessageType.Text),
+                MessageTestUtils.CreateMessage("A", MessageType.Close),
+                MessageTestUtils.CreateMessage("Server Error", MessageType.Error)
+            };
+
+            var output = new ArrayOutput(chunkSize: 8); // Use small chunks to test Advance/Enlarge and partial payload writing
+            foreach (var message in messages)
+            {
+                Assert.True(MessageFormatter.TryWriteMessage(message, output, MessageFormat.Text));
+            }
+
+            Assert.Equal(expectedEncoding, Encoding.UTF8.GetString(output.ToArray()));
+        }
+
+        [Theory]
+        [InlineData("0:B:;", new byte[0])]
+        [InlineData("8:B:q83vEg==;", new byte[] { 0xAB, 0xCD, 0xEF, 0x12 })]
+        [InlineData("8:B:q83vEjQ=;", new byte[] { 0xAB, 0xCD, 0xEF, 0x12, 0x34 })]
+        [InlineData("8:B:q83vEjRW;", new byte[] { 0xAB, 0xCD, 0xEF, 0x12, 0x34, 0x56 })]
```

Comment:
Turns out that's not really well covered in the Base64 API :(. https://github.com/dotnet/corefxlab/issues/1238

It either silently ignores the characters, or throws an `IndexOutOfRangeException`. I don't think I want to encode that behavior in our tests :).